04.ls/wc.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = parse_options
+  input_files = if ARGV.any?
+                  ARGV.map do |file|
+                    {
+                      name: file,
+                      text: File.read(file)
+                    }
+                  end
+                else
+                  [
+                    {
+                      name: '',
+                      text: $stdin.read
+                    }
+                  ]
+                end
```

Comment:
好みの問題なのでMUSTではないですが、ちょっと縦に長すぎるのでこれぐらいにしたいですね。

```suggestion
  input_files = if ARGV.any?
                  ARGV.map do |file|
                    {name: file, text: File.read(file)}
                  end
                else
                  [{ name: '', text: $stdin.read }]
                end
```

さらに言うと、僕は↓のように書くのが好みです。
https://qiita.com/jnchito/items/9b3f476f40803444b471

```suggestion
  input_files =
    if ARGV.any?
      ARGV.map do |file|
        {name: file, text: File.read(file)}
      end
    else
      [{ name: '', text: $stdin.read }]
    end
```